### DISCUSSION TITLE
ERROR= ESP_ERR_NVS_NOT_ENOUGH_SPACE

### DISCUSSION THREAD

**Discussion by loca5790** [emporia-vue-local/esphome#227] [Q&A] [answered]
Hey just saw I needed to change the flash interval to 48 hours after running these things about a year to year and a half.... is this error going to cause flash issues with taking my config?  Or just saving the total energy used?  What I'm getting at is do I need to replace it?

**Comment by flaviut** [comment, marked as answer]
despite my earlier panic, there's no problem. The only possible impact of leaving it on is this specific mis-feature, daily energy being restored after a power outage, may stop working in the future.

Everything else will continue working fine. I'll update the docs.

**Reply by loca5790** [reply]
awesome that's what it looked like when I was digging into the doc's still took flash jsut reset the daily energy, which I'm ok with anyway since I"m tracking it in HA.  Is there a way to force reset total daily energy?

**Reply by flaviut** [reply]
https://esphome.io/components/button/factory_reset.html

**Reply by jcam** [reply]
could you explain what you mean by 'there's no problem'?
I don't want to wear out my flash chip, but i also am seeing that my daily energy numbers are all wrong, seem to be fluctuating all over the place...
<img width="562" alt="image" src="https://github.com/emporia-vue-local/esphome/assets/98744/35fadd0b-a589-47a6-8434-3d9d97766553">

**Reply by flaviut** [reply]
> could you explain what you mean by 'there's no problem'?

The part of the flash that stores the daily energy consumption is used by nothing else. If it wears out, it does not impact the proper operation of the rest of the system.

> but i also am seeing that my daily energy numbers are all wrong, seem to be fluctuating all over the place...

can you please make a new thread?

**Reply by loca5790** [reply]
@flaviut if I add the factory reset button and the NVS not having enough space.... will it still update it so the integration for the home assistant energy monitoring will update correctly on power resets?

**Reply by flaviut** [reply]
I'm sorry, I'm not sure I understand your question. The HA integration will properly handle if the energy consumption resets to zero, even if it isn't midnight, so the NVS restore is never needed.

**Comment by loca5790** [comment]
[E][esp32.preferences:146]: Error saving 19 preferences to flash. Last error=ESP_ERR_NVS_NOT_ENOUGH_SPACE for key=233825507

to combat the above error... have you tried?
```
custom_component:
  - lambda: |-
      ESP_LOGI("main", "Clearing NVS...");
      if (nvs_flash_erase() == ESP_OK) {
          ESP_LOGI("main", "NVS cleared successfully.");
      } else {
          ESP_LOGE("main", "Failed to clear NVS.");
      }
```

I've also been curious about changing partition size as the code only takes up about 42% of the vailable flash.
